src/connection/GRPCHandler.ts

```diff
@@ -1,119 +1,224 @@
-            stubImplementationCodeRequest: call.request
-        });
-        this._factory.get(mes.messageType).process(mes)
-            .then((res) => callback(null, res.fileDiff as gauge.messages.FileDiff))
-            .catch((err) => callback(err, undefined));
-    };
-
-    public validateStep(call: any, callback: gauge.messages.lspService.ValidateStepCallback) {
-        let mes = new gauge.messages.Message({
-            messageId: 0,
-            messageType: gauge.messages.Message.MessageType.StepValidateRequest,
-            stepValidateRequest: call.request
-        });
-        this._factory.get(mes.messageType).process(mes)
-            .then((res) => callback(null, res.stepValidateResponse as gauge.messages.StepValidateResponse))
-            .catch((err) => callback(err, undefined));
-    };
-
-    public refactor(call: any, callback: gauge.messages.lspService.RefactorCallback) {
-        let mes = new gauge.messages.Message({
-            messageId: 0,
-            messageType: gauge.messages.Message.MessageType.RefactorRequest,
-            refactorRequest: call.request
-        });
-        this._factory.get(mes.messageType).process(mes)
-            .then((res) => callback(null, res.refactorResponse as gauge.messages.RefactorResponse))
-            .catch((err) => callback(err, undefined));
-    };
-
-    public getStepName(call: any, callback: gauge.messages.lspService.GetStepNameCallback) {
-        let mes = new gauge.messages.Message({
-            messageId: 0,
-            messageType: gauge.messages.Message.MessageType.StepNameRequest,
-            stepNameRequest: call.request
-        });
-        this._factory.get(mes.messageType).process(mes)
-            .then((res) => callback(null, res.stepNameResponse as gauge.messages.StepNameResponse))
-            .catch((err) => callback(err, undefined));
-    };
-
-    public getGlobPatterns(call: any, callback: gauge.messages.lspService.GetGlobPatternsCallback) {
-        let mes = new gauge.messages.Message({
-            messageId: 0,
-            messageType: gauge.messages.Message.MessageType.ImplementationFileGlobPatternRequest,
-            implementationFileGlobPatternRequest: call.request
-        });
-        this._factory.get(mes.messageType).process(mes)
-            .then((res) => callback(null, res.implementationFileGlobPatternResponse as gauge.messages.ImplementationFileGlobPatternResponse))
-            .catch((err) => callback(err, undefined));
-    };
-
-    public killProcess(call: any, callback: gauge.messages.lspService.KillProcessCallback): void {
-        this._server && this._server.forceShutdown();
-        callback(null, new gauge.messages.Empty());
-        process.exit(0);
-    };
+
+  private readonly _server: Server | null;
+  private readonly _factory: MessageProcessorFactory;
+
+  constructor(server: Server | null, factory: MessageProcessorFactory) {
+    this._server = server;
+    this._factory = factory;
+  }
+
+  public getStepNames(
```

Comment:
Please avoid line breaks in the method definition. It reduces the readability

Reply:
I don't really agree with that. This line break is here especially because of the increase the readability and to avoid long method definitions.

Original definition looked like this

`publig getStepNames(call: GRPCHandlerCallType, callback: gauge.messages.lspService.GetStepNamesCallback): void { `
this definition is too long IMHO.

So from my perspective this definition

`public getStepNames(
    call: GRPCHandlerCallType,
    callback: gauge.messages.lspService.GetStepNamesCallback
): void {`

is more readable. It also provides a developer option to reconsider this definition and easily determine if there are more then 7 (or even less, based on configuration) allowed parameters, so developers can create its own wrapper for these parameters.